src/AbstractSingleton.jl

```diff
@@ -109,6 +110,25 @@ function vertices_list(S::AbstractSingleton{N}
     return [element(S)]
 end
 
+"""
+    linear_map(M::AbstractMatrix, S::AbstractSingleton)
+
+Concrete linear map of an abstract singleton.
+
+### Input
+
+- `M` -- matrix
+- `S` -- abstract singleton
+
+### Output
+
+The abstract singleton of the same type of ``S`` obtained by applying the
+linear map to the element in ``S``.
+"""
+function linear_map(M::AbstractMatrix, S::AbstractSingleton)
+    T = typeof(S)
+    return T(M * element(S))
```

Comment:
Where are the unit tests? :stuck_out_tongue_closed_eyes:

This will not work for `ZeroSet`. And if we ever implement a `OneSet` or whatever, it will also fail. We should at least have a custom implementation for `ZeroSet` for now. And maybe we should change `AbstractSingleton` to `Singleton` here (but I am not sure).
(We might run into this issue more often if we only work with abstract types. I did not see this issue before.)

Reply:
ah, good catch! then let's implement the custom implementation for `ZeroSet`, and leave this one generic as a "fallback"